### PR TITLE
PP-9984 Switch PSP > Update org details - bug creating Sentry error

### PR DESCRIPTION
**Pull request by iqbalgds**
- Update `post controller`
  - Pass `isSwitchingCredentials` flag correctly when getting the Stripe account id.
  - Add unit tests to check this.
  - General clean up of the unit test.

### CODE REVIEW

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 1]
```diff
@@ -177,7 +179,7 @@ describe('organisation address post controller', () => {
       })
     })
 
-    describe('view page when `Stripe setup`', () => {
+    describe('view page when `setting up a new Stripe gateway account`', () => {
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:182]: Can we simplify descriptions as it is confusing to understand the tested functionality?

It is a `post-op` and `view page` in the description implies it is a get. There is another suite `Setup new Stripe account` which looks similar to this. Can these be merged?

Noticed that some tests here are starting with `it('when the required`. Can these be updated for readability? Could be something like:

`it('should return errors, when the required fields are missing')`

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 2]
```diff
@@ -590,7 +593,8 @@ describe('organisation address post controller', () => {
       const mockServiceService = { updateService: mockUpdateService }
       const controller = getController(mockServiceService)
 
-      it('should update connector flag, update Stripe, log message then redirect to `Stripe > Add PSP account details` redirect', async function () {
+      it('should get the stripe account id directly from the account, update connector flag, ' +
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:596]: Similarly, could simplify description and if there are multiple and different things being tested, could split into different tests

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 3]
```diff
@@ -681,7 +688,8 @@ describe('organisation address post controller', () => {
       const mockServiceService = { updateService: mockUpdateService }
       const controller = getController(mockServiceService)
 
-      it('should update connector flag, update Stripe, log message then redirect to `Stripe > Add PSP account details` redirect', async function () {
+      it('should get the stripe account id from the switching credential, update connector flag, ' +
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:691]: Same as above

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 2 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: Please see above

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 3 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: Please see above

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 4]
```diff
@@ -104,26 +107,27 @@ describe('organisation address post controller', () => {
...
-        const responseData = mockResponse.getCalls()[0]
+      responseData = mockResponse.getCalls()[0]
+    }
 
+    describe('request to go live', () => {
+      before(() => {
+        testSuiteSetup({ route: {path: '/request-to-go-live/organisation-address'}})
+      })
+
+      it('when required fields have not been populated, should display errors', () => {
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:130]: Can we re-order this so it reads naturally?

```suggestion
      it('should return errors when the required fields are missing', () => {
```

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 5]
```diff
@@ -133,32 +137,22 @@ describe('organisation address post controller', () => {
...
-
-        controller(req, res, next)
-
-        const responseData = mockResponse.getCalls()[0]
+    describe('manage organisation details', () => {
+      before(() => {
+        testSuiteSetup( { route: { path: '/organisation-details'}})
+      })
 
+      it('when required fields have not been populated, should display errors', () => {
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:155]: Same as above

```suggestion
      it('should return errors when the required fields are missing', () => {
```

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 6]
```diff
@@ -169,30 +163,22 @@ describe('organisation address post controller', () => {
...
-
-        controller(req, res, next)
-
-        const responseData = mockResponse.getCalls()[0]
+    describe('new Stripe gateway account`', () => {
+      before(() => {
+        testSuiteSetup({url: '/your-psp/:credentialId/update-organisation-details'})
+      })
 
+      it('when required fields have not been populated, should display errors', () => {
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:181]:
```suggestion
      it('should return errors when the required fields are missing', () => {
```

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js [hunk 7]
```diff
@@ -201,30 +187,22 @@ describe('organisation address post controller', () => {
...
-
-        controller(req, res, next)
-
-        const responseData = mockResponse.getCalls()[0]
+    describe('`Switch PSP > Stripe`', () => {
+      before(() => {
+        testSuiteSetup({url: '/switch-psp/:credentialId/update-organisation-details'})
+      })
 
+      it('when required fields have not been populated, should display errors', () => {
```
Comment (kbottla) [inline app/controllers/request-to-go-live/organisation-address/post.controller.test.js:205]:
```suggestion
      it('should return errors when the required fields are missing', () => {
```

**Review by kbottla — CHANGES_REQUESTED** [review]

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 4 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: Good point - done

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 5 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: done

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 6 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: done

File: app/controllers/request-to-go-live/organisation-address/post.controller.test.js (hunk 7 above)
Comment (iqbalgds) [reply app/controllers/request-to-go-live/organisation-address/post.controller.test.js]: done

**Review by kbottla — APPROVED** [review]

**Pull request merged**